### PR TITLE
Jenkinsfile: use Ubuntu 20.04 for DCO stage

### PR DESCRIPTION
**Pull request by thaJeztah**
Also switching to use arm64, as all amd64 stages have moved to GitHub actions,
so using arm64 allows the same machine to be used for tests after the DCO check
completed.

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Review by samuelkarp — APPROVED** [review]
LGTM

**Review by AkihiroSuda — APPROVED** [review]

**Comment by tianon** [comment]
LGTM, but we probably want to somehow get this one green for sure before we merge it, right?  (given it's a much heavier change and thus potentially going to cause even more headache if it happens to mask some worse issue further in the pipeline than the flakes?)

**Comment by thaJeztah** [comment]
docker-py failure;

```
=================================== FAILURES ===================================
___________ CreateContainerTest.test_create_with_device_cgroup_rules ___________
tests/integration/api_container_test.py:472: in test_create_with_device_cgroup_rules
    assert rule in self.client.logs(ctnr).decode('utf-8')
E   AssertionError: assert 'c 7:128 rwm' in "cat: can't open '/sys/fs/cgroup/devices/devices.list': No such file or directory\n"
E    +  where "cat: can't open '/sys/fs/cgroup/devices/devices.list': No such file or directory\n" = <built-in method decode of bytes object at 0x7f0efe9cdf30>('utf-8')
E    +    where <built-in method decode of bytes object at 0x7f0efe9cdf30> = b"cat: can't open '/sys/fs/cgroup/devices/devices.list': No such file or directory\n".decode
E    +      where b"cat: can't open '/sys/fs/cgroup/devices/devices.list': No such file or directory\n" = <bound method ContainerApiMixin.logs of <docker.api.client.APIClient object at 0x7f0efe64e390>>({'Id': '7e09d15d63dec23a83bac8f4c2740dc946fcdf2b28fdc2791fcaa2aa7eea37ef', 'Warnings': []})
E    +        where <bound method ContainerApiMixin.logs of <docker.api.client.APIClient object at 0x7f0efe64e390>> = <docker.api.client.APIClient object at 0x7f0efe64e390>.logs
E    +          where <docker.api.client.APIClient object at 0x7f0efe64e390> = <tests.integration.api_container_test.CreateContainerTest testMethod=test_create_with_device_cgroup_rules>.client
```

Test looks like:

```python
@requires_api_version('1.28')
def test_create_with_device_cgroup_rules(self):
    rule = 'c 7:128 rwm'
    ctnr = self.client.create_container(
        TEST_IMG, 'cat /sys/fs/cgroup/devices/devices.list',
        host_config=self.client.create_host_config(
            device_cgroup_rules=[rule]
        )
    )
    self.tmp_containers.append(ctnr)
    config = self.client.inspect_container(ctnr)
    assert config['HostConfig']['DeviceCgroupRules'] == [rule]
    self.client.start(ctnr)
    assert rule in self.client.logs(ctnr).decode('utf-8')
```

With

```python
TEST_IMG = 'alpine:3.10'
```



```bash
docker create --name mycontainer --device-cgroup-rule 'c 7:128 rwm' alpine:3.10 cat /sys/fs/cgroup/devices/devices.list
docker inspect --format '{{json .HostConfig.DeviceCgroupRules }}' mycontainer
["c 7:128 rwm"]

docker start mycontainer
```

On ubuntu 20.04 (kernel 5.4);

```bash
docker logs mycontainer

b *:* m
c *:* m
c 1:3 rwm
c 1:5 rwm
c 1:7 rwm
c 1:8 rwm
c 1:9 rwm
c 5:0 rwm
c 5:1 rwm
c 5:2 rwm
c 7:128 rwm
c 10:200 rwm
c 136:* rwm
```

On Ubuntu 21.10;

```
Cgroup Driver: systemd
 Cgroup Version: 2
...
Kernel Version: 5.13.0-20-generic
```

```bash
docker logs mycontainer
cat: can't open '/sys/fs/cgroup/devices/devices.list': No such file or directory
```

**Comment by thaJeztah** [comment]
^^ following up to the above, as I went looking if there's a replacement for cgroups v2, but for cgroups v2, it's all managed through ebpf and there's no "file" api; https://github.com/torvalds/linux/blob/v5.16/Documentation/admin-guide/cgroup-v2.rst#device-controller

> (...)
> Device controller manages access to device files. It includes both creation of new device files (using mknod), and access to the existing device files.
>
> Cgroup v2 device controller has no interface files and is implemented on top of cgroup BPF. To control access to device files, a user may create bpf programs of type BPF_PROG_TYPE_CGROUP_DEVICE and attach them to cgroups with BPF_CGROUP_DEVICE flag. (...)


I'll add a "skip" in our CI, and will open a PR or ticket in docker-py to fix the test.

**Comment by thaJeztah** [comment]
Rebased this to include https://github.com/moby/moby/pull/43168 - temporarily moving it back to draft until that's merged

**Comment by thaJeztah** [comment]
rebased, and moved out of draft again because https://github.com/moby/moby/pull/43168 was merged

**Review by tianon — APPROVED** [review]

**Comment by thaJeztah** [comment]
I rebased this PR, and it's now only changing the DCO check. As the only architecture we run on PRs in Jenkins is arm64, I modified it to use `arm64` as well, so that the same machine can be re-used to run the tests after the DCO check finished.

I'll merge this one once CI completes 👍

**Pull request merged**
